src/reducers/arcReducer.ts

```diff
@@ -333,6 +334,26 @@ const arcReducer = (state = initialState, action: any) => {
         }
       });
     }
+
+    case ActionTypes.ARC_ON_PROPOSAL_EXECUTED: {
+      const { avatarAddress, proposalId, executionState, decision, reputationWhenExecuted } = payload;
+
+      if (executionState === ExecutionState.PreBoostedBarCrossed || executionState === ExecutionState.BoostedBarCrossed) {
+        return update(state, {
+          proposals: {
+            [proposalId]: {
+              $merge: {
+                reputationWhenExecuted,
+                winningVote: decision,
+                state: ProposalStates.Executed
```

Comment:
Does anything else need to be updated?

Reply:
votesYes and votesNo on the proposal? actually that should be handled by the vote event. also once redeeming is in there could be new redemptions for the current user. ill have to do that. hmm, cant think of anything now